wait/host_port_test.go

```diff
@@ -456,55 +463,166 @@ func TestHostPortStrategyFailsWhileInternalCheckingDueToUnexpectedContainerStatu
 
 func TestHostPortStrategySucceedsGivenShellIsNotInstalled(t *testing.T) {
 	listener, err := net.Listen("tcp", "localhost:0")
-	if err != nil {
-		t.Fatal(err)
-	}
+	require.NoError(t, err)
 	defer listener.Close()
 
 	rawPort := listener.Addr().(*net.TCPAddr).Port
 	port, err := nat.NewPort("tcp", strconv.Itoa(rawPort))
-	if err != nil {
-		t.Fatal(err)
-	}
+	require.NoError(t, err)
 
-	target := &MockStrategyTarget{
-		HostImpl: func(_ context.Context) (string, error) {
-			return "localhost", nil
-		},
-		InspectImpl: func(_ context.Context) (*types.ContainerJSON, error) {
-			return &types.ContainerJSON{
-				NetworkSettings: &types.NetworkSettings{
-					NetworkSettingsBase: types.NetworkSettingsBase{
-						Ports: nat.PortMap{
-							"80": []nat.PortBinding{
-								{
-									HostIP:   "0.0.0.0",
-									HostPort: port.Port(),
-								},
-							},
+	target := &mockStrategyTarget{}
+	target.EXPECT().Host(anyContext).Return("localhost", nil)
+	target.EXPECT().Inspect(anyContext).Return(&types.ContainerJSON{
+		NetworkSettings: &types.NetworkSettings{
+			NetworkSettingsBase: types.NetworkSettingsBase{
+				Ports: nat.PortMap{
+					"80": []nat.PortBinding{
+						{
+							HostIP:   "0.0.0.0",
+							HostPort: port.Port(),
 						},
 					},
 				},
-			}, nil
-		},
-		MappedPortImpl: func(_ context.Context, _ nat.Port) (nat.Port, error) {
-			return port, nil
-		},
-		StateImpl: func(_ context.Context) (*types.ContainerState, error) {
-			return &types.ContainerState{
-				Running: true,
-			}, nil
+			},
 		},
-		ExecImpl: func(_ context.Context, _ []string, _ ...exec.ProcessOption) (int, io.Reader, error) {
-			// This is the error that would be returned if the shell is not installed.
-			return 126, nil, nil
+	}, nil)
+	target.EXPECT().MappedPort(anyContext, nat.Port("80")).Return(port, nil)
```

Comment:
suggestion: no need to do all the tests that already in the other PR just the new tests your adding

Reply:
I'll remove this once other conversations are resolved.